### PR TITLE
Rename provider

### PR DESCRIPTION
**Pull request by mevansam**
Pull request addressing issues: #44 and #66

### CODE REVIEW

File: website/docs/r/user_provided_service.html.markdown [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-layout: "cf"
-page_title: "Cloud Foundry: cf_user_provided_service"
+layout: "cloudfoundry"
+page_title: "Cloud Foundry: cloudfoundry_user_provided_service"
 sidebar_current: "docs-cf-resource-user-provided-service"
```
Comment (gberche-orange) [inline website/docs/r/user_provided_service.html.markdown]: I assume the sidebar_current ids are defined in [cf.erb](https://github.com/mevansam/terraform-provider-cf/blob/dev/website/cf.erb) and therefore don't need renames

Comment (mevansam) [reply website/docs/r/user_provided_service.html.markdown]: For backward compatibility would it make sense to register providers under the cf_* name a well as cloudfoundry_*, until Janos has moved over to the new naming convention? Theoretically, it should work but I have not tested it but wanted to validate if this has already been experimented with at Orange.

**Comment by janosbinder** [comment]
@gberche-orange @mevansam I think the necessary steps for migration are:

0, back up the _tf_ files and the _terraform.tfstate_ file
1, in all _tf_ files replace **resource/data "cf_..."** with **"cloudfoundry_..."**. For example `sed 's#resource \"cf_#resource \"cloudfoundry_#g' old.tf | sed 's#data \"cf_#data \"cloudfoundry_#g'`
2, in the _terraform.tfstate_ file replace the types. For example **"type": "cf_service_instance"** should be replaced with **"type": "cloudfoundry_service_instance"**
3, in the _terraform.tfstate_ file check the _depends_on_ nodes and make the adjustments. For example

> "depends_on": [
>     "data.cf_service.abc"
>     "cf_service_instance.xyz"
>     "cf_route.example"
> ],

Should be 

> "depends_on": [
>     "data.cloudfoundry_service.abc"
>     "cloudfoundry_service_instance.xyz"
>     "cloudfoundry_route.example"
> ],

4, if there are no errors after calling _terraform plan_, and no changes are planned to be applied then the migration was successful.

File: website/docs/r/user_provided_service.html.markdown (hunk 1 above)
Comment (gberche-orange) [reply website/docs/r/user_provided_service.html.markdown]: @mevansam I would suggest to keep the provider code simple and compliant to hashicorp guidelines, and rather rely on beta users of mevansam/terraform-provider-cf to upgrade themselves using the procedure @janosbinder kindly contributed.

Comment (ArthurHlt) [reply website/docs/r/user_provided_service.html.markdown]: I do agree with @gberche-orange this is more complexity for us to maintain than for user to do the changes. In add, we will never remove it to ensure compatibility by doing. I think that the sooner the better here.

**Pull request merged**